tests/flow/test_stress.py

```diff
@@ -113,59 +114,45 @@ def test01_bgsave_stress(self):
         conn = self.env.getConnection()
         graphs[0].query("CREATE INDEX ON :Node(val)")
 
-        t1 = threading.Thread(target=create_nodes, args=(graphs[0], n_iterations, n_nodes))
-        t1.setDaemon(True)
+        pool = Pool(nodes=5)
 
-        t2 = threading.Thread(target=delete_nodes, args=(graphs[1], n_iterations, n_nodes/2))
-        t2.setDaemon(True)
+        t1 = pool.apipe(create_nodes, graphs[0], n_iterations, n_nodes)
 
-        t3 = threading.Thread(target=read_nodes, args=(graphs[2], n_iterations))
-        t3.setDaemon(True)
+        t2 = pool.apipe(delete_nodes, graphs[1], n_iterations, n_nodes/2)
 
-        t4 = threading.Thread(target=update_nodes(graphs[3], n_iterations))
-        t4.setDaemon(True)
+        t3 = pool.apipe(read_nodes, graphs[2], n_iterations)
 
-        t5 = threading.Thread(target=BGSAVE_loop, args=(self.env, conn, 3))
-        t5.setDaemon(True)
+        t4 = pool.apipe(update_nodes, graphs[3], n_iterations)
 
-        t1.start()
-        t2.start()
-        t3.start()
-        t4.start()
-        t5.start()
+        t5 = pool.apipe(BGSAVE_loop, self.env, conn, 3)
 
-        # wait for threads to join
-        t1.join()
-        t2.join()
-        t3.join()
-        t4.join()
-        t5.join()
+        # wait for processes to join
+        t1.wait()
+        t2.wait()
+        t3.wait()
+        t4.wait()
+        t5.wait()
```

Comment:
is there a function within `pool` that would wait for all of the outstanding async operations ?

Reply:
no